src/components/Home.jsx

```diff
@@ -18,80 +23,162 @@ const Home = () => {
   const { topic } = useParams();
   const [articles, SetArticles] = useState([]);
   const [pageNum, setPageNum] = useState(1);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const [_, setTrigger] = useState(false);
+  const { isLoading, setIsLoading } = useContext(LoadingContext);
+
+  const sortByQuery = searchParams.get("sort_by");
+  const orderQuery = searchParams.get("order");
+
+  console.log(orderQuery);
 
   useEffect(() => {
-    getAllArticles(topic, pageNum).then((articles) => {
+    setIsLoading(true);
+    getAllArticles(topic, pageNum, sortByQuery, orderQuery).then((articles) => {
       SetArticles(articles);
       setPageNum(1);
+      setIsLoading(false);
     });
-  }, [topic]);
+  }, [topic, sortByQuery, orderQuery]);
 
   useEffect(() => {
+    setIsLoading(true);
     getAllArticles(topic, pageNum).then((articles) => {
       SetArticles(articles);
+      setIsLoading(false);
     });
   }, [pageNum]);
 
   const changePage = (Num) => {
     setPageNum(Num);
   };
```

Comment:
You have two useEffects here which are overriding each other depending on what is happing and they are both doing most the same thing so they should be combined into one. This would also allow your webpage to use the search params in your url to actually sort by correctly on refresh instead of being reset.